### PR TITLE
feat(tooltip): tooltip formatter trigger once

### PR DESCRIPTION
**Pull request by wf123537200**
## Brief Information

add config as follow to control tooltip.formatter function trigger only once or not
```
tooltip: {
                triggerOnce: true,        // 如果配置为true，formatter函数只触发一次
                ...
            },
```

This pull request is in the type of:

- [ ] bug fixing
- [x] new feature
- [ ] others

### What does this PR do?

Control whether the tooltip.formatter function is triggered only once through configuration items, used to reduce the number of high-cost function executions

### Fixed issues

#12338 

### Before: What was the problem?
the formatter function trigger many times

image

### After: How is it fixed in this PR?
when set triggerOnce: true， each formatter function trigger only once

image

### Are there any API changes?

- [x] The API has been changed.

```
tooltip: {
                triggerOnce: true,        // 如果配置为true，formatter函数只触发一次
                ...
            },
```

### Related test cases or examples to use the new APIs

/test/tooltip-formatter-triggerOnce.html

### Merging options

- [ ] Please squash the commits into a single one when merge.

### CODE REVIEW

**Comment by wf123537200** [comment]
@plainheart thinks for the suggest,and sorry about spelling mistake, i will Implemented in the dispose function late.
i think if the user do these in their own code may a little complicated
if set the cache ```true``` the performance should better then set ```false```

**Comment by plainheart** [comment]
Please add a semicolon `;` at each line end to keep the code style consistent.

**Comment by wf123537200** [comment]
> Please add a semicolon `;` at each line end to keep the code style consistent.

Thank you for reminding, already use the ```npm run lint``` check the code style consistent

File: src/component/tooltip/TooltipView.js [hunk 1]
```diff
@@ -586,7 +610,8 @@ export default echarts.extendComponentView({
                 }
             }, this);
             this._ticket = asyncTicket;
-            html = formatter(params, asyncTicket, callback);
+            html = cacheHtml || formatter(params, asyncTicket, callback);
+            this._setTooltipCache(key, html);
```
Comment (plainheart) [inline src/component/tooltip/TooltipView.js:614]: If `triggerOnce` is not enabled, `enableCache` will be false, so there is no need to call `_setTooltipCache`.
We should wrapped this code line with `if`.

File: src/component/tooltip/TooltipView.js [hunk 2]
```diff
@@ -577,6 +591,16 @@ export default echarts.extendComponentView({
...
+            var enableCache = tooltipModel.get('triggerOnce');
+            if (enableCache && params && params[0]) {
+                var p = params[0];
+                key = [p.componentIndex, p.seriesId, p.seriesName, p.dataIndex].join('_');
+                cacheHtml = this._getTooltipCache(key);
+            }
+
             var callback = bind(function (cbTicket, html) {
                 if (cbTicket === this._ticket) {
                     tooltipContent.setContent(html, markers, tooltipModel);
```
Comment (plainheart) [inline src/component/tooltip/TooltipView.js:606]: The case of `callback` in `formatter` also needs to be handled. Add cache here ?

File: src/component/tooltip/TooltipView.js (hunk 1 above)
Comment (plainheart) [reply src/component/tooltip/TooltipView.js:614]: Maybe no need to wrap, I saw that it won't invoke if `key` is not specified in function `_setTooltipCache`.

Comment (wf123537200) [reply src/component/tooltip/TooltipView.js:614]: From a semantic perspective, it is better to wrap

File: src/component/axis/CartesianAxisView.js [hunk 3]
```diff
@@ -109,7 +109,14 @@ var CartesianAxisView = AxisView.extend({
...
 
         var lineStyle = lineStyleModel.getLineStyle();
+        // the showAsAxis will set lineColor to 'transparent' when the viewLabels value is undefind
+        var showAsAxis = splitLineModel.get('showAsAxis');
+        var viewLabels = showAsAxis && axis.getViewLabels();
+
         for (var i = 0; i < ticksCoords.length; i++) {
+            if(viewLabels && viewLabels[i] && !viewLabels[i].formattedLabel) {
+                continue;
+            }
```
Comment (plainheart) [inline src/component/axis/CartesianAxisView.js:119]: It seems that the changes in this file are related to #12487, which should not be here.

**Comment by wf123537200** [comment]
@plainheart 3q 4 reviewed, rollback the no need code in 12487.

File: test/axis-interval-splitLine.html [hunk 4]
```diff
@@ -0,0 +1,421 @@
+<!DOCTYPE html>
```
Comment (plainheart) [inline test/axis-interval-splitLine.html:1]: Maybe `test/axis-interval-splitLine.html` also should be removed.

**Comment by pissang** [comment]
@wf123537200 Sorry for the late review. Some of my thoughts about this pull request:

1. `triggerOnce` should be a more specific name like `formatterCacheResult`. Or it's easy to be mistaken that this tooltip will only show once.
2. params can be a single object when the trigger is `'item'` in https://github.com/apache/incubator-echarts/pull/12504/files#diff-d5fbb7185a3802482df65d3d3a345c70R598
3. The key generation logic may have issues in some cases, for example, the key won't change if only the number of series changes. 
4. Key generation seems to depend on the business logic. So should we add a new option called `formatterCacheKey`? Which is a callback that receives same parameter with formatter and returns a key string.

**Pull request closed without merging**